Makefile

```diff
@@ -435,6 +436,7 @@ CMAKE_SYSTEM_NAME := Windows
 endif
 
 CONFIGURE_FLAGS += --host=$(XHOST_TRIPLE) CC=$(CC_PATH) CXX=$(CXX_PATH)
+CMAKE_FLAGS += -C$(shell pwd)/build/cmake_cache.txt
```

Comment:
this is weird, Cmake fails when cross-compiling (acceptance tests for some reason). This is kind of the accepted answer (mentioned in the failure itself). I need the absolute path as we're running cmake from the temp build dir.

Reply:
Ugh, what a pain! I think we can at least avoid some of the complexity, though, by dropping

```
set(THREADS_PTHREAD_ARG "2" CACHE STRING "Forcibly set by CMakeLists.txt." FORCE)
```

into libroach's CMakeLists.txt instead of introducing the whole cachefile kit and caboodle.

Could you also add a comment linking to https://gitlab.kitware.com/cmake/cmake/issues/16920? Looks like this is fixed in CMake 3.10 and so this workaround should eventually become unnecessary.